TunnelVisionLabs.ReferenceAssemblyAnnotator/Program.cs

```diff
@@ -33,43 +32,26 @@ internal static void Main(SuppressibleLoggingHelper? log, string referenceAssemb
-            attributesOfInterest.Add(maybeNullAttribute.FullName, maybeNullAttribute);
-            attributesOfInterest.Add(maybeNullWhenAttribute.FullName, maybeNullWhenAttribute);
-            attributesOfInterest.Add(notNullAttribute.FullName, notNullAttribute);
-            attributesOfInterest.Add(notNullIfNotNullAttribute.FullName, notNullIfNotNullAttribute);
-            attributesOfInterest.Add(notNullWhenAttribute.FullName, notNullWhenAttribute);
+            attributesOfInterest.Add(wellKnownTypes.SystemRuntimeCompilerServicesNullableAttribute.Value.FullName, wellKnownTypes.SystemRuntimeCompilerServicesNullableAttribute.Value.Resolve());
+            attributesOfInterest.Add(wellKnownTypes.SystemRuntimeCompilerServicesNullableContextAttribute.Value.FullName, wellKnownTypes.SystemRuntimeCompilerServicesNullableContextAttribute.Value.Resolve());
+            attributesOfInterest.Add(wellKnownTypes.SystemRuntimeCompilerServicesNullablePublicOnlyAttribute.Value.FullName, wellKnownTypes.SystemRuntimeCompilerServicesNullablePublicOnlyAttribute.Value.Resolve());
+            attributesOfInterest.Add(wellKnownTypes.SystemDiagnosticsCodeAnalysisAllowNullAttribute.Value.FullName, wellKnownTypes.SystemDiagnosticsCodeAnalysisAllowNullAttribute.Value.Resolve());
+            attributesOfInterest.Add(wellKnownTypes.SystemDiagnosticsCodeAnalysisDisallowNullAttribute.Value.FullName, wellKnownTypes.SystemDiagnosticsCodeAnalysisDisallowNullAttribute.Value.Resolve());
+            attributesOfInterest.Add(wellKnownTypes.SystemDiagnosticsCodeAnalysisDoesNotReturnAttribute.Value.FullName, wellKnownTypes.SystemDiagnosticsCodeAnalysisDoesNotReturnAttribute.Value.Resolve());
+            attributesOfInterest.Add(wellKnownTypes.SystemDiagnosticsCodeAnalysisDoesNotReturnIfAttribute.Value.FullName, wellKnownTypes.SystemDiagnosticsCodeAnalysisDoesNotReturnIfAttribute.Value.Resolve());
+            attributesOfInterest.Add(wellKnownTypes.SystemDiagnosticsCodeAnalysisMaybeNullAttribute.Value.FullName, wellKnownTypes.SystemDiagnosticsCodeAnalysisMaybeNullAttribute.Value.Resolve());
+            attributesOfInterest.Add(wellKnownTypes.SystemDiagnosticsCodeAnalysisMaybeNullWhenAttribute.Value.FullName, wellKnownTypes.SystemDiagnosticsCodeAnalysisMaybeNullWhenAttribute.Value.Resolve());
+            attributesOfInterest.Add(wellKnownTypes.SystemDiagnosticsCodeAnalysisNotNullAttribute.Value.FullName, wellKnownTypes.SystemDiagnosticsCodeAnalysisNotNullAttribute.Value.Resolve());
+            attributesOfInterest.Add(wellKnownTypes.SystemDiagnosticsCodeAnalysisNotNullIfNotNullAttribute.Value.FullName, wellKnownTypes.SystemDiagnosticsCodeAnalysisNotNullIfNotNullAttribute.Value.Resolve());
+            attributesOfInterest.Add(wellKnownTypes.SystemDiagnosticsCodeAnalysisNotNullWhenAttribute.Value.FullName, wellKnownTypes.SystemDiagnosticsCodeAnalysisNotNullWhenAttribute.Value.Resolve());
```

Comment:
What do you think of:

```cs
foreach (var attribute in new[]
{
    wellKnownTypes.SystemRuntimeCompilerServicesNullableAttribute,
    wellKnownTypes.SystemRuntimeCompilerServicesNullableContextAttribute,
    ...
})
{
    attributesOfInterest.Add(attribute.Value.FullName, attribute.Resolve());
}
```

Reply:
I was actually thinking of just extracting a local function to add the items.

Reply:
That works too! Or foreach over `wellKnownTypes.AllNullabilityAttributes`.

Reply:
Now simplified

Reply:
Or `wellKnownTypes.AllNullabilityAttributes.ToDictionary(a => a.Value.FullName, a.Value.Resolve());`